### ISSUE TITLE
[Question] How can I display the name of the current collection in a listview page?

### ISSUE THREAD

**Issue by BFallert** [kitodo/kitodo-presentation#918]
When a collection is clicked on a page that contains a collection list (plugin: Collection), a normal ListView is called.

Currently, however, it is not visible in this ListView page that it was called via the link of a collection.
In the contained Typo3 variables it can be recognized via `lastSearch[collection]` that the template was called via a collection link, but instead of the collection name only the collection number is contained (i.e. the ID in the Typo3 database).

Is there a possibility to output the data of the collection list, especially **display label, description, thumbnail,** also in the ListView of a collection?

**Comment by michaelkubina** [comment]
The Listview-Controller provides such information via the `currentCollection` object. In the code it is even assigned to the view, but from what i see it does not get applied anywhere in the template ([\Resources\Private\Templates\ListView\Main.html](https://github.com/kitodo/kitodo-presentation/blob/master/Resources/Private/Templates/ListView/Main.html)) by default.

It would be required add something like the snippet shown below to the template - but with a lot collections this information block might grow huge. And of course it needs to be styled in your CSS somehow.

I can only speak for certain that it works for the current master branch (not the last release). The current master has already the changes to the collection-plugin that i wrote. This being said i am not sure if the last release already returned a list of collections or just a single item. There were several severe bugs, that made collections not reliable or functional back then.

**\Resources\Private\Templates\ListView\Main.html**
```
...
            <f:for each="{currentCollection}" as="currentCollectionItem">
                <div class="collection">
                    <div class="collection-label">
                        <f:if condition="{currentCollectionItem.thumbnail}">
                            <f:image image="{currentCollectionItem.thumbnail}" />
                        </f:if>
                    <div>
                    <div class="collection-thumbnail">
                        <f:if condition="{currentCollectionItem.label}">
                            {currentCollectionItem.label}
                        </f:if>
                    </div>
                    <div class="collection-description">
                        <f:if condition="{currentCollectionItem.description}">
                            {currentCollectionItem.description}
                        </f:if>
                    </div>
                </div>
            </f:for>
...
```

**Comment by BFallert** [comment]
I am working with the 4.x branch, but `currentCollection` does not exist.
`currentCollection` is not found anywhere in the source code? 
Main-Branch and 4.x do not differ significantly.
Have your changes been incorporated elsewhere?
Am I looking in the wrong place?

**Comment by michaelkubina** [comment]
You are right, i am sorry! I have mistaken a local adjustment as part of the actual branch. You are right, there is no `currentCollection` in any branch. If you want to output the collection information, then you need to pass the already resolved collection(s) to your view in the [ListViewController.php](https://github.com/kitodo/kitodo-presentation/blob/master/Classes/Controller/ListViewController.php)

`$this->view->assign('currentCollection', $collection);`

Then call it in the template as i stated above...

If you consider it an expected feature, it would be worth discussing if it should be an option in the listview-plugin.

**Comment by BFallert** [comment]
Thanks a lot! That helped me!

For me, it would be an expected feature of the ListView plugin.

It would be a simple straightforward way to display the data entered to a collection. I can put together a PR, then feel free to discuss it there.

**Comment by BFallert** [comment]
Just a few small corrections in case someone wants to use this as example

```
...
            <f:for each="{currentCollection}" as="currentCollectionItem">
                <div class="collection">
                    <div class="collection-thumbnail">
                        <f:if condition="{currentCollectionItem.thumbnail}">
                            <f:image image="{currentCollectionItem.thumbnail}" />
                        </f:if>
                    </div>
                    <div class="collection-label">
                        <f:if condition="{currentCollectionItem.label}">
                            {currentCollectionItem.label}
                        </f:if>
                    </div>
                    <div class="collection-description">
                        <f:if condition="{currentCollectionItem.description}">
                            <f:format.html>{currentCollectionItem.description}</f:format.html>
                        </f:if>
                    </div>
                </div>
            </f:for>
...
```

**Comment by sebastian-meyer** [comment]
There are two different templates for the ListView in Kitodo.Presentation:

- Resources/Private/Templates/ListView/Main.html
- Resources/Private/Templates/Collection/List.html

The former is used as a default for any kind of list, that does not have its own template assigned. This template does not contain any information about collections.
The latter is used when ListView is called from within the Collection controller. This template handles collection information assigned by the Collection controller and displays those above the result list.

(At least this is how it should be working. If it's not, that's a bug...)
